dbms/src/Functions/repeat.cpp

```diff
@@ -0,0 +1,236 @@
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Functions/castTypeToEither.h>
+
+namespace DB
+{
+	namespace ErrorCodes
+	{
+		extern const int ILLEGAL_COLUMN;
+		extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+	}
+
+	struct RepeatImpl
+	{
+        static void vectorNonConstStr(
+            const ColumnString::Chars & data,
+            const ColumnString::Offsets & offsets,
+            ColumnString::Chars & res_data,
+            ColumnString::Offsets & res_offsets,
+            const UInt64 repeatTime)
+        {
+            UInt64 data_size = 0;
+            res_offsets.assign(offsets);
+            for (UInt64 i = 0; i < offsets.size(); ++i)
+			{
+                data_size += (offsets[i] - offsets[i - 1] - 1) * repeatTime + 1;
```

Comment:
BTW, we can calculate data_size in single formula out of this loop

Reply:
But we also need to calculate res_offsets one by one.